content/tutorial_next_step/02_adding_modules.en.md

```diff
@@ -0,0 +1,271 @@
+---
+title: Adding modules
+---
+
+## Adding modules
+
+It does not take long before you need to split up your state and signals. In Cerebral you can use modules to handle that. Modules are a very simple concept, they basically namespace state, signals and services to give structure. Let us see how we can define a module for our application:
+
+[src/model.js](https://github.com/cerebral/cerebral-website-tutorial-next/blob/react/src/controller.js)
+
+```javascript
+import {Controller} from 'cerebral'
+import Model from 'cerebral/models/immutable'
+import Devtools from 'cerebral-module-devtools'
+import Http from 'cerebral-module-http'
+import App from './modules/App'
+
+const controller = Controller(Model({}))
+
+controller.addModules({
+  app: App,
+
+  http: Http({
+    baseURL: '/api'
+  }),
+  devtools: Devtools()
+})
+
+export default controller
+```
+
+What we have done now is add a namespace to our application called **app** and we reference our App module. We also have two other modules, **devtools** and **http**. That means there is no difference creating an application specific module or a module you want to share with others. Let us take a closer look at our **App** module:
+
+[src/modules/App/index.js](https://github.com/cerebral/cerebral-website-tutorial-next/blob/react/src/modules/App/index.js)
+
+```javascript
```

Comment:
is it from https://github.com/cerebral/cerebral-website-tutorial-next/blob/react/src/modules/App/index.js ?
don't inline it manually.
I will do it in build step in future.